vm/src/obj/objstr.rs

```diff
@@ -519,6 +519,88 @@ impl PyString {
         }
     }
 
+    // Return true if all characters in the string are printable or the string is empty,
+    // false otherwise.  Nonprintable characters are those characters defined in the
+    // Unicode character database as `Other` or `Separator`,
+    // excepting the ASCII space (0x20) which is considered printable.
+    //
+    // All characters except those characters defined in the Unicode character
+    // database as following categories are considered printable.
+    //   * Cc (Other, Control)
+    //   * Cf (Other, Format)
+    //   * Cs (Other, Surrogate)
+    //   * Co (Other, Private Use)
+    //   * Cn (Other, Not Assigned)
+    //   * Zl Separator, Line ('\u2028', LINE SEPARATOR)
+    //   * Zp Separator, Paragraph ('\u2029', PARAGRAPH SEPARATOR)
+    //   * Zs (Separator, Space) other than ASCII space('\x20').
+    #[pymethod]
+    fn isprintable(&self, _vm: &VirtualMachine) -> bool {
+        self.value.is_empty()
+            || self.value.chars().all(|c| {
+                match c {
+                    // Other, Control
+                    '\u{0000}'| '\u{0001}'| '\u{0002}'| '\u{0003}'| '\u{0004}'| '\u{0005}'|
```

Comment:
The should exist a rust library which has these tables implemented. For example this library: https://crates.io/crates/unicode_categories